cTraining/cTraining/Game.c

```diff
@@ -0,0 +1,166 @@
+   StartGame ();
+   //AutomationTest ();
+}
+
+int MarkSymbols (int cellNumber, char board[][ROWCOUNT], char sym) {
+   int row = cellNumber <= 3 ? 0 : cellNumber <= 6 ? 1 : 2;
+   int col = cellNumber <= 3 ? cellNumber - 1 : cellNumber <= 6 ? cellNumber - 4 : cellNumber - 7;
+   if (isdigit (board[row][col])) {
+      board[row][col] = sym;
+      return 1;
+   }
+   return 0;
+}
+
+int CheckWin (char board[][ROWCOUNT], char sym) {
+   for (int i = 0; i < ROWCOUNT; i++) {
+      //checking columns and rows
+      if ((board[i][0] == sym && board[i][1] == sym && board[i][2] == sym) ||
+         (board[0][i] == sym && board[1][i] == sym && board[2][i] == sym))
+         return 1;
+   }
+   //checking diagonals
+   if ((board[0][2] == sym && board[1][1] == sym && board[2][0] == sym) ||
+      (board[0][0] == sym && board[1][1] == sym && board[2][2] == sym))
+      return 1;
+   return 0;
+}
+
+void InitBoard (char board[][ROWCOUNT]) {
+   printf ("\n");
+   for (int i = 0; i < ROWCOUNT; i++) {
+      for (int j = 0; j < ROWCOUNT; j++)
+         printf ("|%c", board[i][j]);
+      printf ("|\n");
+   }
+}
+
+void Input (int* player) {
+   char playerInput;
+   do
+      playerInput = getch ();
+   while (!(playerInput >= '1' && playerInput <= '9'));
+   *player = playerInput - '0';
+}
+
+void StartGame () {
+   printf ("\nPlayers can enter a number between 1 to 9.\nPlayer 1's symbol is 'O'\nPlayer 2's symbol is 'X'\n");
+   char board[][ROWCOUNT] = { { '1','2','3' },{ '4','5','6' },{ '7','8','9' } }, sym;
+   int cellNumber, it = 0, ret, i = 0;
+   InitBoard (board);
+   while (true) {
+      sym = i ? 'X' : 'O';
+      it++;
+      do {
+         printf ("\nPlayer%d:", i + 1);
+         Input (&cellNumber);
+         ret = MarkSymbols (cellNumber, board, sym);
+         if (!ret) printf ("Position is occupied!\n");
+      } while (!ret);
+      InitBoard (board);
+      if (it >= 5) {
+         ret = CheckWin (board, sym);
+         if (ret) {
+            printf ("\nPlayer%d is win\n", i + 1);
+            break;
+         } else if (it == ROWCOLUMNCOUNT) {
+            printf ("\nMatch draw!");
+            break;
+         }
+      }
+      i++;
+      if (i > 1) i = 0;
+   }
+}
+
+int AutomationTest () {
+   int input1[] = { 1,2,3,4,5,6,7,8,9 }, input2[] = { 2,3,1,4,6,8,7,5,9 };
+   int input3[] = { 1,2,2,1,3,4,3,5,6 };
+   char expOut1[][ROWCOUNT] = { { 'O','X','O' },{ 'X','O','X' },{ 'O','X','O' } };
+   char expOut2[][ROWCOUNT] = { { 'O','O','X' },{ 'X','X','O' },{ 'O','X','O' } };
+   char expOut3[][ROWCOUNT] = { {'O','X','O' },{ 'X','X','O' },{ '7','8','9' } };
+   if (!TestMarkSymbols (input1, expOut1) || !TestMarkSymbols (input2, expOut2) || !TestMarkSymbols (input3, expOut3)) return 0;
+   if (!TestCheckWin (expOut1, 'O', 1) || !TestCheckWin (expOut2, 'X', 0) || !TestCheckWin (expOut3, 'O', 0)) return 0;
+   printf ("Test cases: Passed!\n");
+   return 1;
+}
+
+int TestMarkSymbols (int input[], char expOut[][ROWCOUNT]) {

```

Comment:
Why is this specification required here? Seems to serve no purpose, no error is shown if you change the array size in the calling code

```suggestion
int TestMarkSymbols (int input[], char expOut[][]) {

```

Reply:
![Screenshot 2025-01-20 084230](https://github.com/user-attachments/assets/6fca9882-ce22-46e3-88a0-a0bffe6d3137)